packages/openapi-v3/src/parameter-decorator.ts

```diff
@@ -0,0 +1,411 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  isSchemaObject,
+  ParameterObject,
+  ParameterLocation,
+  ReferenceObject,
+  SchemaObject,
+} from '@loopback/openapi-v3-types';
+import {MetadataInspector, ParameterDecoratorFactory} from '@loopback/context';
+import {getSchemaForParam} from './';
+import {OAI3Keys} from './keys';
+
+/**
+ * Describe an input parameter of a Controller method.
+ *
+ * `@param` must be applied to parameters. For example,
+ * ```
+ * class MyController {
+ *   @get('/')
+ *   list(
+ *     @param(offsetSpec) offset?: number,
+ *     @param(pageSizeSpec) pageSize?: number,
+ *   ) {}
+ * }
+ * ```
+ *
+ * @param paramSpec Parameter specification.
+ */
+export function param(paramSpec: ParameterObject) {
+  return function(target: Object, member: string | symbol, index: number) {
+    paramSpec = paramSpec || {};
+    // Get the design time method parameter metadata
+    const methodSig = MetadataInspector.getDesignTypeForMethod(target, member);
```

Comment:
Copying because github ignored my comment: 
I think this is a remnant of having method level `@param` decorator. Can we rework the code here so that we don't need leverage `getDesignTypeForMethod` here?

Reply:
see comment https://github.com/strongloop/loopback-next/pull/1046#discussion_r171049244